features/build.js

```diff
@@ -0,0 +1,360 @@
+                cellContentWriter.write(marked.parse(synopsis));
+                empty = false;
+              }
+              if (empty) {
+                cellContentWriter.write('&nbsp;');
+              }
+            });
+
+            // API Documentation
+            rowWriter.class(`px-1 ${commonCellStyle}`);
+            rowWriter.cell((cellContentWriter) => {
+              cellContentWriter.write(apiDocumentation ? marked.parse(apiDocumentation) : '&nbsp;');
+            });
+
+            // SDK columns
+            sdkManifests.forEach((manifest) => {
+              const compliance = manifest.find([...parentKeys, key]);
+
+              let colourClass = 'bg-red-400';
+              let svg = crossSvg;
+              if (compliance) {
+                const { variants, notes } = compliance;
+                const hasPartialSupportForVariants = variants && manifest.isPartialVariantsCoverage(variants);
+                const hasPartialSupportNotes = !!notes;
+                if (hasPartialSupportForVariants || hasPartialSupportNotes) {
+                  colourClass = 'bg-amber-400';
+                  svg = partialSvg;
+                } else {
+                  colourClass = 'bg-green-400';
+                  svg = tickSvg;
+                }
+              }
+
+              rowWriter.class(`px-1 ${colourClass} ${commonCellStyle}`);
+              rowWriter.cell((cellContentWriter) => {
+                cellContentWriter.write(svg);
+              });
+            });
+          });
+        }
+
+        const depth = generateTableRows(writer, maximumLevel, [...parentKeys, key], value);
+        maximumDepth = Math.max(maximumDepth, 1 + depth);
+      }
+    });
+  } else if (Array.isArray(node)) {
+    node.forEach((element) => {
+      const depth = generateTableRows(writer, maximumLevel, parentKeys, element);
+      maximumDepth = Math.max(maximumDepth, depth);
+    });
+  } else if (node instanceof String || typeof node === 'string') {
+    if (writer) {
+      console.log(`${consoleIndent}"${node}"`);
+    }
+    maximumDepth = 1;
+  } else if (node === null) {
+    // the value for a key with no value defined
+  } else if (writer) {
+    // informational only, while debugging
+    console.log(`${consoleIndent}${typeof node} = ${node}`);
+  }
+
+  return maximumDepth;
+}
+
+/**
+ * Creates a directory at the given path if it doesn't exist, recursively if necessary.
+ *
+ * @param {string} directoryPath The directory path. Can be relative to current working directory.
+ */
+function createDirectory(directoryPath) {
+  if (!fs.existsSync(directoryPath)) {
+    fs.mkdirSync(directoryPath, { recursive: true });
+  }
+}
+
+/**
+ * Inspects YAML AST to ensure source constraints are met (e.g. ordered keys).
+ *
+ * @param {*} astNode The YAML AST node.
+ * @param {number} level The depth, for recursion depth protection.
+ */
+function validateStructure(astNode, level = 0) {
+  if (astNode === null) {
```

Comment:
What is the legitimate use case where a node can be `null`?

Reply:
I've added commentary, but have also refactored the code to move the `null` check to where it really should have been (narrower scope of acceptance) in https://github.com/ably/ably-common/pull/142/commits/8832445d6143ec9896487bcf83705a81bfce63b2.